src/__test__/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapLegendVisibilitySettings.test.jsx

```diff
@@ -0,0 +1,166 @@
+import configureMockStore from 'redux-mock-store';
+import { Radio } from 'antd';
+
+import { UPDATE_CONFIG } from '../../../../../../../redux/actionTypes/componentConfig';
+import HeatmapLegendVisibilitySettings from '../../../../../../../pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapLegendVisibilitySettings';
+
+const initialState = {
+  cellSets: {
+    // cellSets', 'metadataCategorical
+    hierarchy: [
+      {
+        key: 'louvain',
+        children: [
+          {
+            key: 'louvain-0',
+          },
+          {
+            key: 'louvain-1',
+          },
+        ],
+      },
+      {
+        key: 'sample',
+        children: [
+          {
+            key: 'control',
+          },
+          {
+            key: 'treated',
+          },
+        ],
+      },
+    ],
+    properties: {
+      louvain: {
+        type: 'cellSets',
+        name: 'louvain clusters',
+      },
+      'louvain-0': {
+        name: 'louvain 0',
+        cellIds: [5, 6, 7],
+      },
+      'louvain-1': {
+        name: 'louvain 1',
+        cellIds: [1, 2, 3],
+      },
+      sample: {
+        type: 'metadataCategorical',
+        name: 'Sample',
+      },
+      control: {
+        name: 'control',
+        cellIds: [5, 6, 7],
+      },
+      treated: {
+        name: 'treated',
+        cellIds: [1, 2, 3],
+      },
+    },
+  },
+  cellInfo: {},
+  componentConfig: {
+    interactiveHeatmap: {
+      config: {
+        groupedTrack: 'sample',
+        selectedTracks: ['louvain'],
+        legendIsVisible: true,
+      },
+    },
+  },
+};
+
+const mockStore = configureMockStore([thunk]);
+configure({ adapter: new Adapter() });
+
+const eventStub = {
+  stopPropagation: () => { },
+};
+
+describe('HeatmapLegendVisibilitySettings', () => {
+  let component, store, buttons;
+
+  afterEach(() => {
+    component.unmount();
+  });
+
+  beforeEach(() => {
+    store = mockStore({
+      ...initialState,
+    });
+
+    component = mount(
+      <Provider store={store}>
+        {/* ask if experimentId='123' width={200} height={200} are necessary in HeatmapGroupBySettings */}
+        <HeatmapLegendVisibilitySettings experimentId='123' width={200} height={200} />
+      </Provider>,
+    );
+  });
+
+  it('renders correctly', () => {
+    const buttons = component.find('Radio');
+
+    // Should be rendered.
+    expect(component.find('HeatmapLegendVisibilitySettings').length).toEqual(1);
+
+    // With two buttons.
+    expect(buttons.length).toEqual(2);
+
+    // With the right keys.
+    expect(buttons.at(0).props().value).toEqual(true);
+    expect(buttons.at(1).props().value).toEqual(false);
+
+    // The selected one should be `Show`.
+    expect(component.find(Radio.Group).props().value).toEqual(true);
+  });
+
+  it('responds correctly to clicking to hide', () => {
+    // Click hide button
+    act(() => {
+      component.find('ForwardRef').props().onChange({ "target": { "value": false } });
```

Comment:
I tried using simulate() with the 'change' event on this which seems cleaner, but it wouldn't work on the Radio elements  so I went for this option, if you believe there is a better was for me to do this please let me know.
@ivababukova @marcellp

Reply:
did you try to simulate a `click` event, as it is done in here: https://github.com/biomage-ltd/ui/blob/master/src/__test__/components/FocusButton.test.jsx#L98

Reply:
yes, initially I tried with click, but it doesnt work on this particular element.

Reply:
That is correct, Radio buttons are a pain to work with.

I think I had a conversation with Vicky about the implementation. My understanding was that it would not be a radio button but rather just the menu option showing `Show legend` and `Hide legend`, but I see this was not done in this implementation. I assume the communication chain broke at some point, but it's okay, for the purposes of the feature this will do.

Reply:
so we leave it with the radio button?

I was just looking at your new tests in your differential expression PR by the way and saw that you could also use find(Radio.Button) which is more declarative, so I'll change it for that.